File: src/main/java/com/NovumScientiaTeam/novumscientia/api/materialProperties/transfer/EnergyTransferProperties.java
package com.NovumScientiaTeam.novumscientia.api.materialProperties.transfer;

import com.EmosewaPixel.pixellib.materialSystem.materials.Material;

import java.util.HashMap;
import java.util.Map;

public class EnergyTransferProperties extends TransferProperties {
    private static final Map<Material, EnergyTransferProperties> energyTransferMap = new HashMap<>();

    public static void setMaterialProperties(Material mat, int itemCapacity, int itemsPerTick) {
        energyTransferMap.put(mat, new EnergyTransferProperties(itemCapacity, itemsPerTick));
    }

    public static EnergyTransferProperties getPropertiesForMaterial(Material mat) {
        return energyTransferMap.get(mat);
    }

    private EnergyTransferProperties(int energyCapacity, int FEPerTick) {
        super(energyCapacity, FEPerTick);
    }
}

File: src/main/java/com/NovumScientiaTeam/novumscientia/api/materialProperties/transfer/EnergyTransferProperties.kt
package com.novumscientiateam.novumscientia.api.materialproperties.transfer

import com.emosewapixel.pixellib.materialsystem.materials.Material

import java.util.HashMap

class EnergyTransferProperties private constructor(energyCapacity: Int, FEPerTick: Int) : TransferProperties(energyCapacity, FEPerTick) {
    companion object {
        private val energyTransferMap = HashMap<Material, EnergyTransferProperties>()

        fun setMaterialProperties(mat: Material, itemCapacity: Int, itemsPerTick: Int) {
            energyTransferMap[mat] = EnergyTransferProperties(itemCapacity, itemsPerTick)
        }

        fun getPropertiesForMaterial(mat: Material) = energyTransferMap[mat]
    }
}